transaction-status/src/option_serializer.rs

```diff
@@ -0,0 +1,78 @@
+use serde::{ser::Error, Deserialize, Deserializer, Serialize, Serializer};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum OptionSerializer<T: Default> {
+    Some(T),
+    None,
+    Skip,
+}
+
+impl<T: Default> Default for OptionSerializer<T> {
+    fn default() -> Self {
+        Self::None
+    }
+}
+
+impl<T: Default> OptionSerializer<T> {
+    pub fn skip() -> Self {
+        Self::Skip
+    }
+
+    pub fn should_skip(&self) -> bool {
+        matches!(self, Self::Skip)
+    }
+
+    pub fn or_skip(option: Option<T>) -> Self {
+        match option {
+            Option::Some(item) => Self::Some(item),
+            Option::None => Self::Skip,
+        }
+    }
+
+    pub fn as_ref(&self) -> Option<&T> {
```

Comment:
I would expect this to return `Self<&T>`

Reply:
Fair, cutting dumb corners here. Fixed.